Packs/CommonScripts/Scripts/HttpV2/HttpV2.py

```diff
@@ -0,0 +1,206 @@
+    if request_content_type_header and not headers.get('Content-Type'):
+        headers['Content-Type'] = request_content_type_header
+    if response_content_type_header and not headers.get('Accept'):
+        headers['Accept'] = response_content_type_header
+
+    return headers
+
+
+def get_parsed_response(res, resp_type: str) -> Dict:
+    try:
+        resp_type = resp_type.lower()
+        if resp_type == 'json':
+            res = res.json()
+        elif resp_type == 'text':
+            res = res.text
+        elif resp_type == 'xml':
+            res = json.loads(xml2json(res.content))
+        else:
+            res = res.text
+        return res
+    except ValueError as exception:
+        raise DemistoException('Failed to parse json object from response: {}'
+                               .format(res.content), exception)
+
+
+def get_status_list(status_list: list) -> List[int]:
+    """
+    Args:
+        status_list: The given status list.
+    Returns:
+        A list of statuses to retry.
+    """
+    final_status_list = []
+    for status in status_list:
+        range_numbers = status.split('-')
+        if len(range_numbers) == 1:
+            final_status_list.append(int(range_numbers[0]))
+        else:
+            status_range = list(range(int(range_numbers[0]), int(range_numbers[1]) + 1))
+            final_status_list.extend(status_range)
+    return final_status_list
+
+
+def build_outputs(parsed_res, res) -> Dict:
+    return {'ParsedBody': parsed_res,
+            'Body': res.text,
+            'StatusCode': res.status_code,
+            'StatusText': res.reason,
+            'URL': res.url,
+            'Headers': dict(res.headers)}
+
+
+''' MAIN FUNCTION '''
+
+
+def main(args: Dict):
+    method = args.get('method', '')
+    full_url = args.get('url', '')
+    body = args.get('body', '')
+    request_content_type = args.get('request_content_type', '')
+    response_content_type = args.get('response_content_type', '')
+    parse_response_as = args.get('parse_response_as', RAW_RESPONSE)
+    params = args.get('params', {})
+    headers = args.get('headers', {})
+    headers = create_headers(headers, request_content_type, response_content_type)
+    auth = tuple(argToList(args.get('auth_credentials', None)))
+    save_as_file = args.get('save_as_file', 'no')
+    file_name = args.get('filename', 'http-file')
+    enable_redirect = argToBoolean(args.get('enable_redirect', True))
+    timeout = arg_to_number(args.get('timeout', 10))
+    retry_on_status = args.get('retry_on_status', None)
+    raise_on_status = True if retry_on_status else False
+    retry_status_list = get_status_list(argToList(retry_on_status))
+    timeout_between_retries = args.get('timeout_between_retries', 5)
+    retry_count = arg_to_number(args.get('retry_count', 3))
+    proxy = argToBoolean(args.get('proxy', False))
+    verify = argToBoolean(not args.get('unsecure', False))
```

Comment:
Let's try to access and parse each argument as close to where it's used as possible